cosima_cookbook/querying.py

```diff
@@ -244,6 +245,20 @@ def _preprocess(d):
     for attr in variables[1:]:
         da.attrs[attr] = ds[attr]
 
+    # Get experiment metadata, delete extraneous fields and add 
+    # to attributes
+    metadata = get_experiments(session, 
+                               experiment=False, 
+                               exptname=expt, 
+                               all=True).to_dict(orient='records')[0]
+    map(metadata.__delitem__, ['ncfiles', 'index', 'root_dir'])
```

Comment:
I don't think this does anything unless you actually iterate over the returned `map` object, does it? The tests aren't catching this anyway. This (and the following block) could be replaced by a dictionary comprehension, e.g. (formatting aside, I haven't actually tried this):

```python
metadata = {
    k: v for k, v in metadata.items()
    if k not in ["ncfiles", "index", "root_dir"]
    and (v is not None and v != "None" and v != "")
}
da.attrs.update(metadata)
```

Reply:
Seems to work, so I'm not sure, but I agree your comprehension is a nice solution. 

Ninja comprehension FTW.

Thanks.

Reply:
Hmm, doesn't work here:

```
>>> x = {"a": 1, "b": 2}
>>> map(x.__delitem__, ["a"])
<map object at 0x7fb7226e2910>
>>> x
{'a': 1, 'b': 2}
```

Reply:
Yeah, when I tested it I did
```
list(map(x.__delitem__, ["a"]))
```
which might have sent me down the wrong path .. thanks